### ISSUE TITLE
Relationships are able to cross tenants with attribute strategy

### ISSUE THREAD

**Issue by rbino** [ash-project/ash_postgres#192]
**Describe the bug**
Ash Postgres currently allows relating two resources from two different tenants when using attribute strategy and referencing the primary key of the other resource.

**To Reproduce**
 [Here](https://github.com/rbino/ash_postgres/commit/d8f62872e94bb8e7e115d7e24ec715142ffe4142#diff-ee89cab7243923108e4ded2df70b2b46b17b4acbedfadf03af83c35944d4a04cR24) is a commit that adds two tests that manage cross-tenant relationships.

The one that references the primary key is able to add a cross-tenant relationship, the other correctly fails (with a rather cryptic error message though, so I'm not sure if it's actually failing for the right reason).

**Expected behavior**
I would expect cross-tenant relationship not to be possible

**Runtime**
 - Elixir version: 1.15.4
 - Erlang version: 26.0.2
 - OS: Debian Testing
 - Ash Postgres version: `main`

**Additional context**
The [Ecto guide](https://hexdocs.pm/ecto/multi-tenancy-with-foreign-keys.html#working-with-multi-tenant-associations) covers this usecase by adding composite foreign keys to disallow inserting cross-tenant associations.

This is what happens for relationships referencing non-primary key attributes, and it was happening also for primary key attributes before 0adec1d1634387367bb818e40711e34ceb8cb3e4.

That commit was made to fix the error shown in #144, but I think that the fix shouldn't have been dropping the composite foreign key, but rather adding the extra `[:id, :org_id]` unique index in the migration generator (should this be added for all multitenant-attribute resources or just as soon as their foreign key gets referenced for the first time?), as is also shown in the Ecto guide.

Of course the cross-tenant relationship is correctly filtered out when reading the resource so it can't be retrieved, but I think it would be better to disallow its creation completely at the database level given there's the possibility to do so.

**Comment by zachdaniel** [comment]
Yeah, I think that is a good idea 👍Will be a breaking change though, so will need some level of configuration backing it and/or to wait for a major release.

**Comment by TravelCurry02** [comment]
I have been doing some spelunking on this issue. Before any fixes to multitenant attribute relationships, Ash tried to make a foreign key that included the tenant column, even when it pointed at another table's primary key. That caused migrations to fail in Postgres, which led to issue #144: a foreign key on (id, tenant) needs a unique index on (id, tenant), but it only had uniqueness on id. The fix that was implemented changed it so that if the relationship points at the primary key, it does not add the tenant column to the foreign key. That allowed migrations to work again, but it also made it possible to create relationships across tenants. Since restoring the composite foreign key plus unique index would be a breaking change, would you rather make it an optional setting for now, or make the composite foreign key plus unique index the default in a major release?

**Comment by zachdaniel** [comment]
Let's make it an optional setting for now. I'm not so sure that we need to use the primary key, I think what we'd do is add a `with` that matches the tenant to the target resource's tenant.
